GUI/CloneFakeKspDialog.cs

```diff
@@ -0,0 +1,243 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace CKAN
+{
+    /// <summary>
+    /// The GUI implementation of clone and fake.
+    /// It's a seperate window, handling the whole process.
+    /// </summary>
+    public partial class CloneFakeKspDialog : Form
+    {
+        private GUIUser user = new GUIUser();
+        private KSPManager manager;
+
+        private ChooseKSPInstance chooseKSPInstanceForm;
+
+        public CloneFakeKspDialog(KSPManager manager, ChooseKSPInstance chooseKSPInstanceForm)
+        {
+            this.manager = manager;
+            this.chooseKSPInstanceForm = chooseKSPInstanceForm;
+
+            InitializeComponent();
+
+            // Populate the combobox for fake instance.
+            List<Versioning.KspVersion> knownVersions = new GameVersionProviders.KspBuildMap(new Win32Registry()).KnownVersions;
+            knownVersions.Reverse();
+            comboBoxKspVersion.DataSource = knownVersions;
+        }
+
+        #region clone
+
+        /// <summary>
+        /// The KSP object for the instance the user selected to clone.
+        /// Fills the path textbox automatically.
+        /// </summary>
+        private KSP selectedInstance
+        {
+            set
+            {
+                _selectedInstance = value;
+                textBoxClonePath.Text = selectedInstance.GameDir();
+            }
+            get { return _selectedInstance; }
+        }
+        private KSP _selectedInstance;
+
+        /// <summary>
+        /// Click event for the OpenInstanceSelection button, which is used to raise a selection dialog
+        /// to choose which known KSP instance the user wants to clone.
+        /// </summary>
+        private void buttonOpenInstanceSelection_Click(object sender, EventArgs e)
+        {
+            // Get all to the regisrty known instances.
+            KSP[] knownInstances = manager.Instances.Values.ToArray();
+            List<string> instancesAsStrings = new List<string>();
+            
+            // Now turn them into a list of nice, readable strings.
+            foreach (KSP instance in knownInstances)
+            {
+                instancesAsStrings.Add(String.Format("{0} ({1}) at {2}", instance.Name, instance.Version().ToString(), instance.GameDir()));
+            }
+            
+            // Raise the selection dialog.
+            int selection = user.RaiseSelectionDialog("Choose an existing instance:", instancesAsStrings.ToArray());
+
+            // Now set selectedInstance to the picked one.
+            selectedInstance = knownInstances[selection];            
```

Comment:
If the user cancels out of the popup, won't `selection` be -1, causing this to crash?